### PR TITLE
Cleanup endpoint timeout and compression settings

### PR DESCRIPTION
**Pull request by pepone**
This PR cleanups the compression and timeout settings, the corresponding public properties are removed as the NewTimeout/NewCompressionFlag methods used to change the settings.

`CreateIPEndpoint` is replaced by `CloneWithHostAndPort`

### CODE REVIEW

File: csharp/src/Ice/TcpEndpoint.cs [hunk 1]
```diff
@@ -13,12 +13,13 @@ namespace ZeroC.Ice
     /// <summary>The Endpoint class for the TCP transport.</summary>
     internal class TcpEndpoint : IPEndpoint
     {
-        public override bool HasCompressionFlag { get; }
         public override bool IsDatagram => false;
         public override bool IsSecure => Transport == Transport.SSL;
-        public override TimeSpan Timeout { get; } = DefaultTimeout;
         public override Transport Transport { get; }
 
+        protected readonly bool HasCompressionFlag;
```
Comment (bernardnormier) [inline csharp/src/Ice/TcpEndpoint.cs:20]: I would keep them as get-only `private protected` properties.

File: csharp/src/Ice/UdpEndpoint.cs [hunk 2]
```diff
@@ -44,7 +43,7 @@ public override bool Equals(Endpoint? other)
                 return true;
             }
             return other is UdpEndpoint udpEndpoint &&
-                HasCompressionFlag == udpEndpoint.HasCompressionFlag &&
+                _hasCompressionFlag == udpEndpoint._hasCompressionFlag &&
```
Comment (bernardnormier) [inline csharp/src/Ice/UdpEndpoint.cs:46]: I think keeping `_hasCompressionFlag` in Equals and GetHashCode was an oversight in Benoit's PR.

File: csharp/src/Ice/Endpoint.cs [hunk 3]
```diff
@@ -51,11 +45,8 @@ public abstract class Endpoint : IEquatable<Endpoint>
                 {
                     return option switch
                     {
-                        "compress" => HasCompressionFlag ? "true" : null,
                         "host" => Host.Length > 0 ? Host : null,
```
Comment (bernardnormier) [inline csharp/src/Ice/Endpoint.cs:48]: compress and timeout should move to TcpEndpoint and UdpEndpoint, not disappear entirely.

File: csharp/test/Ice/info/AllTests.cs [hunk 4]
```diff
@@ -39,11 +39,6 @@ public static void Run(TestHelper helper)
                 TestHelper.Assert(tcpEndpoint.Host == "tcphost");
                 TestHelper.Assert(tcpEndpoint.Port == 10000);
                 TestHelper.Assert(tcpEndpoint["source-address"] == "10.10.10.10");
-                if (ice1)
```
Comment (bernardnormier) [inline csharp/test/Ice/info/AllTests.cs:42]: You could keep these tests.

File: csharp/src/Ice/IPEndpoint.cs [hunk 5]
```diff
@@ -356,11 +351,6 @@ private protected IPEndpoint(InputStream istr, Communicator communicator, Protoc
 
         private protected abstract IConnector CreateConnector(EndPoint addr, INetworkProxy? proxy);
 
-        // TODO: rename to Clone and make parameters optional?
-        private protected abstract IPEndpoint CreateIPEndpoint(
-            string host,
-            ushort port,
-            bool compressionFlag,
-            TimeSpan timeout);
+        private protected abstract IPEndpoint CloneWithHostAndPort(string host, ushort port);
```
Comment (bernardnormier) [inline csharp/src/Ice/IPEndpoint.cs:354]: Should we call this method simply Clone?

File: csharp/src/Ice/TcpEndpoint.cs [hunk 6]
```diff
@@ -228,19 +226,15 @@ protected internal override void WriteOptions(OutputStream ostr)
         private protected override IConnector CreateConnector(EndPoint addr, INetworkProxy? proxy) =>
             new TcpConnector(this, addr, proxy);
 
-        private protected override IPEndpoint CreateIPEndpoint(
-            string host,
-            ushort port,
-            bool compressionFlag,
-            TimeSpan timeout) =>
+        private protected override IPEndpoint CloneWithHostAndPort(string host, ushort port) =>
```
Comment (bernardnormier) [inline csharp/src/Ice/TcpEndpoint.cs:229]: Should Clone return this if host == Host and port == Port?

File: csharp/src/Ice/TcpEndpoint.cs (hunk 1 above)
Comment (pepone) [reply csharp/src/Ice/TcpEndpoint.cs:20]: I guess it should be private, because the class is sealed

File: csharp/src/Ice/TcpEndpoint.cs (hunk 6 above)
Comment (pepone) [reply csharp/src/Ice/TcpEndpoint.cs]: Yes. I don't see why not

File: csharp/src/Ice/IPEndpoint.cs (hunk 5 above)
Comment (pepone) [reply csharp/src/Ice/IPEndpoint.cs]: fixed

File: csharp/test/Ice/info/AllTests.cs (hunk 4 above)
Comment (pepone) [reply csharp/test/Ice/info/AllTests.cs]: fixed

File: csharp/src/Ice/Endpoint.cs (hunk 3 above)
Comment (pepone) [reply csharp/src/Ice/Endpoint.cs:48]: Fixed

File: csharp/src/Ice/UdpEndpoint.cs (hunk 2 above)
Comment (pepone) [reply csharp/src/Ice/UdpEndpoint.cs]: Fixed

**Pull request merged**